packages/util/src/bytes.ts

```diff
@@ -225,6 +226,19 @@ export function bigIntToBuffer(num: bigint) {
   return toBuffer('0x' + num.toString(16))
 }
 
+/**
+ * Converts a {@link bigint} to a {@link Buffer} for devp2p rlp encodings
+ * excluding `0` as rlp has its own special encoding for the same
+ * see: https://github.com/ethereumjs/ethereumjs-monorepo/issues/2402
+ */
+export function devP2PBigIntToBuffer(input: bigint | number): BufferLike {
+  const inputNum = BigInt(input)
+  if (inputNum === 0n) {
+    return 0
```

Comment:
```suggestion
    return Buffer.alloc(0)
```

Reply:
Now it always returns a Buffer type.

Note: `RLP.encode(Buffer.alloc(0))` = `<Buffer 80>` (as we expect)